src/lib/download-progress.ts

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash';
+import blink = require('./blink');
+import { DeviceStatus, TargetState } from '../types/state';
+import * as deviceState from '../device-state';
+import * as targetState from '../device-state/target-state';
+
+interface ServiceState {
+	id: string;
+	status: string;
+	download_progress: number | null;
+}
+
+const DOWNLOAD_BLINK_THROTTLE = 10000;
```

Comment:
Any reason for this particular combination of timings? Maybe you could add a comment for future context?

Reply:
No reason other than it looked good. I tried some other combinations and settled on this one eventually as it felt easy to spot. Noted on the comments.